tests/pytests/test_json.py

```diff
@@ -617,8 +617,17 @@ def test_JSON_RDB_load_fail_without_JSON_module(env: Env):
     env.envRunner.modulePath.pop() # Assumes Search module is the first and JSON module is the second
     env.envRunner.moduleArgs.pop()
     env.envRunner.masterCmdArgs = env.envRunner.createCmdArgs('master')
-    env.start() # Restart without JSON module. Attempt to load RDB
-    env.assertFalse(env.isUp()) # Server is down with no assertion error (MOD-7587)
+    # Restart without JSON module. Attempt to load RDB - should fail.
+    # RLTest may or may not fail to start the server with an exception
```

Comment:
Consider changing to:
```suggestion
    # RLTest `start` function assumes it will always manage to bring the process up successfully 
    # It can raise an exception if the process terminates while we are still inside the function
    # From the test perspective we only care that the process terminated and didn't manage to start without the json module
```